yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,264 @@
+)
+
+const (
+	MONITOR_HEALTH_FREQUENCY_SECONDS = 120 // 2 mins interval for monitoring the cluster checks
+	METRIC_FREE_DISK_SPACE           = "free_disk_space"
+	METRIC_TOTAL_DISK_SPACE          = "total_disk_space"
+	FREE_DISK_SPACE_THREASHOLD       = 10
+	REPLICATION_GUARDRAIL_ALERT_MSG  = "It is NOT recommended to have any form of replication (CDC/xCluster) running on the target YugabyteDB cluster during data import. If you still wish to proceed with replication, use the '--skip-replication-checks true' flag to disable the check."
+	NODE_GOES_DOWN_MSG               = "Nodes in the target YugabyteDB cluster are unhealthy, continuing data import using the available healthy nodes. Please check them on the target cluster"
+	NODE_COMES_BACK_UP_MSG           = "Previously unhealthy nodes in the target YugabyteDB cluster have recovered"
+)
+
+type TargetDBForMonitorHealth interface {
+	GetYBServers() (bool, []*tgtdb.TargetConf, error)
+	RemoveConnectionsForHosts([]string) error
+	NumOfLogicalReplicationSlots() (int64, error)
+	GetClusterMetrics() (map[string]tgtdb.NodeMetrics, error) // node_uuid:metric_name:metric_value
+}
+
+type YugabyteDBClient interface {
+	Init() error
+	SetYBServers(string)
+	SetSSLRootCert(string)
+	GetNumOfReplicationStreams() (int, error)
+}
+
+type MonitorTargetYBHealth struct {
+	// key is the target cluster node and value is bool variable for indicating if a node is down or up
+	nodesStatus map[string]bool
+	//ybClient instance to fetch streams
+	ybClient dbzm.YugabyteDBCDCClient
+	//yb tdb instance to query target for monitoring
+	yb TargetDBForMonitorHealth
+
+	//Skip booleans as per flags
+	skipNodeCheck        bool
+	skipDiskCheck        bool
+	skipReplicationCheck bool
+
+	//Display function for the printing any msg on the console
+	displayMsgFunc func(string)
+
+	tconf tgtdb.TargetConf
+}
+
+func NewMonitorTargetYBHealth(yb TargetDBForMonitorHealth, skipDiskUsageHealthChecks utils.BoolStr, skipReplicationChecks utils.BoolStr, skipNodeHealthChecks utils.BoolStr, displayMsgFunc func(msg string), ybClient YugabyteDBClient, tconf tgtdb.TargetConf) MonitorTargetYBHealth {
+	return MonitorTargetYBHealth{
+		nodesStatus:          make(map[string]bool),
+		yb:                   yb,
+		skipNodeCheck:        bool(skipNodeHealthChecks),
+		skipDiskCheck:        bool(skipDiskUsageHealthChecks),
+		skipReplicationCheck: bool(skipReplicationChecks),
+		displayMsgFunc:       displayMsgFunc,
+		tconf:                tconf,
+	}
+}
+
+func (m *MonitorTargetYBHealth) StartMonitoring() error {
+
+	var err error
+	loadBalancerEnabled, servers, err := m.yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	for _, server := range servers {
+		m.nodesStatus[server.Host] = true
+	}
+
+	if loadBalancerEnabled {
+		m.skipNodeCheck = true
+	}
+
+	//skipping the disk usage monitoring for now, util the metrics function changes are made
+	m.skipDiskCheck = true
```

Comment:
nit: instead of hardcoding it here, hardcode it while calling `NewMonitorTargetYBHealth`